### DISCUSSION TITLE
Model Collection not refreshed after attach or detach (many to many)

### DISCUSSION THREAD

**Discussion by hitoridev** [livewire/livewire#4522] [1. Help] [answered]
hi guys, i've problem on refreshing model collection after detach. i want to remove detached item from view without refresh fullpage, but it seem not work. i need to refresh fullpage to rerender the view. I have tried to `$model->refresh()`  after `detach()`, and its work but  `$model->refresh()` duplicate already loaded relationship query.
Here the route:
```php
Route::get('/order/{order}', function (Order $order) {
    $order = $order->load(['worklist.khs:id,mou_id,worklist_id,khs', 'worklist.category:id,category', 'materials']);
    return view('livewire.order.index-order',  ['order' => $order]);
})->name('view.order');
```
Livewire Component:
```php
class ViewOrder extends Component
{
    public Order $order;
    public function delete($id)
    {
        $work = Worklist::find($id);
        $this->order->worklist()->detach($work);
        $this->order->refresh();  //refreshed, but relationship query duplicated
        $this->tempTotal();
    }

public function render()
    {
        return view('livewire.order.view-order', ['worklist' => $this->order->worklist]);
    }
{
```
Screenshoot of duplicated query if use $model->refresh():
![duplicated](https://user-images.githubusercontent.com/72544746/149651891-c694ce8c-34bc-4853-a4cf-5e2a65248100.jpg)

is there any workaround to refresh relationship collection without fullpage refresh?
thanks

**Comment by joshhanley** [comment, marked as answer]
@hitoridev I think you can't avoid the second relation query, as you need to get the latest data from the database after the change happens, so there will be two queries, one before and one after.

I probably wouldn't use `refresh` though as that reloads the model too, which isn't needed, so instead you can use `->load(...)` to just reload a specific relationship.

Hope this helps!

**Reply by hitoridev** [reply]
i have tried your solution, and its work, relation updated in view. but I still get 1 duplication of query. I mean is it ok if query duplicated like that in the real project? does it affect Performance? thanks
![duplicated1](https://user-images.githubusercontent.com/72544746/149852951-66309b24-57de-421d-a091-b5fcfaffc115.jpg)

**Reply by joshhanley** [reply]
@hitoridev yeah once duplication is ok in my opinion when you are reloading changed data 🙂 but everyones apps and requirements are different, so you will need to assess that based on your current traffic load.

**Reply by hitoridev** [reply]
glad to hears that, thank you @joshhanley . Can i ask one more question that bothering me?
why relation query loaded from route `web.php` suddenly changed to `SerializesAndRestoresModelIdentifiers.php` after reloading data, and the query become `SELECT *` from `SELECT 'id', 'column','column'`?

**Reply by joshhanley** [reply]
@hitoridev the issue is Livewire has no way of knowing that you have selected custom columns, so after Livewire does a request, it does a `SELECT *` instead of your custom columns. This is because Livewire only keeps track of the model name and IDs and then uses that to restore the data.

Hope this helps!
